azure/scope/managedcontrolplane.go

```diff
@@ -526,8 +533,15 @@ func (s *ManagedControlPlaneScope) AgentPoolSpec() azure.AgentPoolSpec {
 		replicas = *s.MachinePool.Spec.Replicas
 	}
 
+	var name string
+	if s.InfraMachinePool.Spec.Name != nil {
```

Comment:
can we instead do this in webhook to check for `spec.name`, and default to `metadata.name`? wdyt?

Reply:
make sense to me! updated the webhook w/ defaulting logic